src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/Diagnostics/SqlClientMetrics.cs

```diff
@@ -0,0 +1,346 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics.Tracing;
+#if NET
+using System.Threading;
+#endif
+
+#nullable enable
+
+namespace Microsoft.Data.SqlClient.Diagnostics
+{
+    // This encapsulates three types of metrics:
+    // * Default: These metrics are always enabled.
+    // * Verbose: These metrics aren't enabled by default, but can be enabled on application startup if a trace switch is set to verbose.
+    // * Trace: These metrics are enabled with the SqlClientEventSource.
+    // Verbose metrics are only present for backwards compatibility.
+    internal sealed partial class SqlClientMetrics
+    {
+        private readonly SqlClientEventSource _eventSource;
+
+        public SqlClientMetrics(SqlClientEventSource eventSource)
+        {
+            _eventSource = eventSource;
+            EnableDefaultMetrics();
+        }
+
+        private void EnableDefaultMetrics()
+        {
+#if NETFRAMEWORK
+            EnablePerformanceCounters();
+#endif
+        }
+
+        public void EnableTraceMetrics()
+        {
+#if NET
+            EnableEventCounters();
+#endif
+        }
+
+        /// <summary>
+        /// The number of actual connections that are being made to servers
+        /// </summary>
+        internal void HardConnectRequest()
+        {
+#if NET
+            Interlocked.Increment(ref _activeHardConnectionsCounter);
+            Interlocked.Increment(ref _hardConnectsCounter);
+#endif
+#if NETFRAMEWORK
```

Comment:
There's been some discussion on this, but the way I'm looking at it, if NET isn't set NETFRAMEWORK will be set. I think we can just use an `#else` for these.